sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/WriteGroupedRowsToFiles.java

```diff
@@ -105,13 +108,13 @@ public void processElement(
         }
       }
 
-      List<ManifestFile> manifestFiles =
-          Preconditions.checkNotNull(writer.getManifestFiles().get(windowedDestination));
-      for (ManifestFile manifestFile : manifestFiles) {
+      List<SerializableDataFile> serializableDataFiles =
+          Preconditions.checkNotNull(writer.getSerializableDataFiles().get(windowedDestination));
+      for (SerializableDataFile dataFile : serializableDataFiles) {
```

Comment:
Are there additional serialization/deserialization costs now due to data being re-encoded as PCollection elements instead of being written to files here ?

Reply:
See previous comment (we're not re-encoding the file contents). If you check out the document, there's actually evidence of performance improvements